src/stan/lang/generator/local_var_decl_visgen.hpp

```diff
@@ -168,9 +158,7 @@ namespace stan {
 
       void operator()(const double_var_decl& x) const {
         std::vector<expression> ctor_args;
-        declare_array(is_fun_return_
-                      ? "fun_scalar_t__"
-                      : (is_var_context_ ? "T__" : "double"),
+        declare_array(is_fun_return_ ? "fun_scalar_t__" : "local_scalar_t__",
```

Comment:
Can you just use `local_scalar_t__` in functions instead of or in addition to `fun_scalar_t__`?

Reply:
yes you can.
I think we can take this one step further and get rid of typedef for `fun_scalar_t__` altogether?

Reply:
simplifying generated function body, eliminating "fun_scalar_t__" and using "local_scalar_t__" instead.